egs/hub4_english/s5/run.sh

```diff
@@ -0,0 +1,286 @@
+
+  steps/train_deltas.sh --boost-silence 1.25 --cmd "$train_cmd" 2000 10000 \
+    data/train_2k data/lang_nosp exp/mono0a_ali exp/tri1
+fi
+
+if [ $stage -le 17 ]; then
+  steps/align_si.sh --nj $nj --cmd "$train_cmd" \
+    data/train data/lang_nosp exp/tri1 exp/tri1_ali
+
+  steps/train_lda_mllt.sh --cmd "$train_cmd" 2500 15000 \
+    data/train data/lang_nosp exp/tri1_ali exp/tri2
+fi
+
+if [ $stage -le 18 ]; then
+  steps/align_si.sh --nj $nj --cmd "$train_cmd" \
+    data/train data/lang_nosp exp/tri2 exp/tri2_ali
+
+  steps/train_sat.sh --cmd "$train_cmd" 4200 40000 \
+    data/train data/lang_nosp exp/tri2_ali exp/tri3
+fi
+
+if [ $stage -le 19 ]; then
+  utils/mkgraph.sh data/lang_nosp_test exp/tri3 exp/tri3/graph_nosp
+
+  for dset in $test_sets; do
+    (
+    this_nj=`cat data/$dset/spk2utt | wc -l`
+    if [ $this_nj -gt 20 ]; then
+      this_nj=20
+    fi
+    steps/decode_fmllr.sh --nj $this_nj --cmd "$decode_cmd" --num-threads 4 \
+      exp/tri3/graph_nosp data/$dset exp/tri3/decode_nosp_${dset} || touch exp/tri3/.error
+    steps/lmrescore_const_arpa.sh --cmd "$decode_cmd" \
+      data/lang_nosp_test data/lang_nosp_test_rescore \
+      data/${dset} exp/tri3/decode_nosp_${dset} \
+      exp/tri3/decode_nosp_${dset}_rescore || touch exp/tri3/.error
+    ) &
+  done
+  wait
+
+  if [ -f exp/tri3/.error ]; then
+    echo "Decode failed in exp/tri3/decode*"
+    exit 1
+  fi
+fi
+
+if [ $stage -le 20 ]; then
+  steps/align_fmllr.sh --nj $nj --cmd "$train_cmd" \
+    data/train data/lang_nosp exp/tri3 exp/tri3_ali
+
+  steps/train_sat.sh --cmd "$train_cmd" 5000 100000 \
+    data/train data/lang_nosp exp/tri3_ali exp/tri4
+fi
+
+if [ $stage -le 21 ]; then
+  utils/mkgraph.sh data/lang_nosp_test exp/tri4 exp/tri4/graph_nosp
+
+  for dset in $test_sets; do
+    (
+    this_nj=`cat data/$dset/spk2utt | wc -l`
+    if [ $this_nj -gt 20 ]; then
+      this_nj=20
+    fi
+    steps/decode_fmllr.sh --nj $this_nj --cmd "$decode_cmd" --num-threads 4 \
+      exp/tri4/graph_nosp data/$dset exp/tri4/decode_nosp_${dset}
+    steps/lmrescore_const_arpa.sh --cmd "$decode_cmd" \
+      data/lang_nosp_test data/lang_nosp_test_rescore \
+      data/${dset} exp/tri4/decode_nosp_${dset} \
+      exp/tri4/decode_nosp_${dset}_rescore
+    ) &
+  done
+  wait
+
+  if [ -f exp/tri4/.error ]; then
+    echo "Decode failed in exp/tri4/decode*"
+    exit 1
+  fi
+fi
+
+wait
+
+# %WER 18.0 | 728 32834 | 83.9 11.7 4.3 2.0 18.0 85.9 | exp/tri4/decode_nosp_eval97.pem_rescore/score_14_0.0/eval97.pem.ctm.filt.sys
+# %WER 19.3 | 728 32834 | 82.9 12.6 4.6 2.2 19.3 86.8 | exp/tri4/decode_nosp_eval97.pem/score_13_0.0/eval97.pem.ctm.filt.sys
+
+# The following demonstrates how to use out-of-domain WSJ models to segment long
+# audio recordings of HUB4 with raw unaligned transcripts into short segments
+# with aligned transcripts for training new ASR models.
+
+# First run the data preparation stages in WSJ run.sh
```

Comment:
The script from here seems like it would be better as part of local/run_segmentation_wsj.sh.  E.g. wsj_base could be made anargument to that script.  And all this could be commented out.  Unless I misunderstand something.  I assume this isn't actually needed by the rest of this script, or it would be at the top...?